### PR TITLE
gh-101632: Add a RETURN_CONST instruction

### PR DESCRIPTION
**Pull request by penguin-wwy**
* Issue: gh-101632

### CODE REVIEW

**Comment by iritkatriel** [comment]
Do you see a difference in any benchmarks?

**Comment by penguin-wwy** [comment]
I am executing pyperformance, please wait :)

**Comment by penguin-wwy** [comment]
@iritkatriel I update the pyperformance result in issue
https://github.com/python/cpython/issues/101632#issuecomment-1420353969

**Comment by iritkatriel** [comment]
> @iritkatriel I update the pyperformance result in issue
> [#101632 (comment)](https://github.com/python/cpython/issues/101632#issuecomment-1420353969)

Thanks, I'll try to replicate on our benchmark machine.

**Comment by iritkatriel** [comment]
> > @iritkatriel I update the pyperformance result in issue
> > [#101632 (comment)](https://github.com/python/cpython/issues/101632#issuecomment-1420353969)
> 
> Thanks, I'll try to replicate on our benchmark machine.

Also came out 1% faster.

Results are [here](https://github.com/faster-cpython/benchmarking-public/tree/main/results/bm-20230207-3.12.0a4%2B-f1deb5c).

**Comment by penguin-wwy** [comment]
> > > @iritkatriel I update the pyperformance result in issue
> > > [#101632 (comment)](https://github.com/python/cpython/issues/101632#issuecomment-1420353969)
> > 
> > 
> > Thanks, I'll try to replicate on our benchmark machine.
> 
> Also came out 1% faster.
> 
> Results are [here](https://github.com/faster-cpython/benchmarking-public/tree/main/results/bm-20230207-3.12.0a4%2B-f1deb5c).

Thanks. It turned out a little better than I expected.

File: Lib/test/test_code.py [hunk 1]
```diff
@@ -723,6 +723,7 @@ def f():
             pass
         PY_CODE_LOCATION_INFO_NO_COLUMNS = 13
         f.__code__ = f.__code__.replace(
+            co_stacksize=1,
```
Comment (iritkatriel) [inline Lib/test/test_code.py:726]: Why was this necessary?

Comment (penguin-wwy) [reply Lib/test/test_code.py:726]: The function `f`:
```
def f():
    pass

LOAD_CONST 0 (None) # The stack size is 1
RETURN_VALUE
```

After optimization
```
RETURN_CONST 0 (None) # The stack size is 0
```

So need to specify the `co_stacksize` when `replace` call.

**Review by iritkatriel — APPROVED** [review]
LGTM

**Comment by iritkatriel** [comment]
The "s390x Fedora Refleaks PR" buildbot is broken (for a few months at least). All other test runs were successful.
